app/models/ability.rb

```diff
@@ -0,0 +1,71 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new
+
+    can :manage, :all if user.is_admin?
+    can [:manage_item, :manage_content, :manage_desc_metadata, :register_apo], :all if user.is_manager?
+    can [:view_item, :view_metadata, :view_content], :all if user.is_viewer?
+
+    can :manage_item, Dor::AdminPolicyObject do |dor_item|
+      dor_item.can_manage_item? user.roles(dor_item.pid)
+    end
+
+    can :manage_content, Dor::AdminPolicyObject do |dor_item|
+      dor_item.can_manage_content? user.roles(dor_item.pid)
+    end
+
+    can :manage_desc_metadata, Dor::AdminPolicyObject do |dor_item|
+      dor_item.can_manage_desc_metadata? user.roles(dor_item.pid)
+    end
+
+    can :manage_item, ActiveFedora::Base do |dor_item|
+      return unless dor_item.admin_policy_object
+
+      dor_item.admin_policy_object.can_manage_item? user.roles(dor_item.pid)
+    end
+
+    can :manage_content, ActiveFedora::Base do |dor_item|
+      return unless dor_item.admin_policy_object
+
+      dor_item.admin_policy_object.can_manage_content? user.roles(dor_item.pid)
+    end
+
+    can :manage_desc_metadata, ActiveFedora::Base do |dor_item|
+      return unless dor_item.admin_policy_object
+
+      dor_item.admin_policy_object.can_manage_desc_metadata? user.roles(dor_item.pid)
+    end
+
+    can :view_item, Dor::AdminPolicyObject do |dor_item|
+      dor_item.can_view_item? user.roles(dor_item.pid)
+    end
+
+    can :view_content, Dor::AdminPolicyObject do |dor_item|
+      dor_item.can_view_content? user.roles(dor_item.pid)
+    end
+
+    can :view_desc_metadata, Dor::AdminPolicyObject do |dor_item|
+      dor_item.can_view_desc_metadata? user.roles(dor_item.pid)
+    end
```

Comment:
Is the order of those checks important? It seems likely that it isn't, if we're falling through. If it is, we can simply reverse the order of the `Dor::AdminPolicyObject` and `ActiveFedora::Base` checks.

cc @LynnMcRae ?